pkg/configfilearg/parser.go

```diff
@@ -101,14 +103,32 @@ func (p *Parser) findStart(args []string) ([]string, []string, bool) {
 		return []string{}, args, true
 	}
 
+	afterIndex := make(map[string]int)
+	re, err := regexp.Compile(`:\d`)
+	if err != nil {
+		return args, nil, false
+	}
+	// After keywords ending with ":<NUM>" will set + NUM of arguments as the split point.
+	// used for matching on subcommmands
+	for i, arg := range p.After {
+		if re.MatchString(arg) {
+			split := strings.Split(arg, ":")
```

Comment:
Rather than matching and then splitting, capture the digits from the regular expression with `:(\d+)` and then use that. Saves you a string operation. Although it seems unlikely, `\d+` is probably better than `\d` in case we ever want more than 9 arguments.  Also, should we ensure that this is >0?

Reply:
Reworked for group matching

Reply:
Since developers only write this, I see no need to check for > 0, the unit tests will cover that.